src/services/squidrouter/process.ts

```diff
@@ -40,6 +41,8 @@ export async function squidRouter(
   } catch (e) {
     console.error('Error in squidRouter: ', e);
     return { ...state, failure: 'unrecoverable' };
+  } finally {
+    trackEvent(createTransactionEvent('signing_requested', state));
```

Comment:
That's the thing we could not differentiate between a sign rejection, or an error. We can assume that if we reach the next event it was successful and if not, it was either an error or was rejected.

Reply:
Hmm well now that you mention it, the catch statement returns from this function in l. 43 so the `finally` part will only be reached if the `try` block doesn't fail. Maybe I should duplicate the code and put the call to `trackEvent` below the call to `await writeContract()` ie. between l. 40-41 as well as above the return in l. 42-43. What do you think @gianfra-t?

Reply:
Mmm.. I think the `finally` statement will execute regardless! Not sure if it will do so before or after returning, but it should execute it. in that sense I think it is fine.

Assuming it is tracked on both scenarios, what I meant is that the user may not have seen the request but still it will be equivalent as if it was requested but rejected, judging from our events only. But probably it is threading too thin 😅.

Reply:
The `finally` will always execute. Either after the `return` of the `catch` block before it actually returns or after a successful execution of the `try` block.